### ISSUE TITLE
Fix S2387 FP: Do not raise issue when 'new' modifier is present

### ISSUE THREAD

**Issue by jagat95** [SonarSource/sonar-dotnet#3393]
### Description

https://rules.sonarsource.com/csharp/RSPEC-2387
In the example they have shown standard class inheritance but does it not consider the C# 'new' keyword for hiding parent properties. I understand it could still cause confusion as the bug says but I feel it should be allowed in case we explicitly follow C#'s own pragmatics 

### Repro steps

```
class A
{
 protected readonly BaseClass prop;
}
class B : A
{
 private new readonly ChildClass prop; // bug error line
}

```
`// the BaseClass is another class which is inherited by ChildClass
// prop is private readonly property. 
`
### Expected behavior
I would still like to maintain same property name.

I am expecting that adding **new** keyword tells the class to hide the underlying parent field and get rid of this error

### Actual behavior

Error still persists even with or without **new** keyword

### Related information

- C#
-  7.0
- Visual Studio Version - 2017 Enterprise
- Win 10
- Sonar Qube Developer edition 8.1
- Sonar Lint add on latest

**Comment by pavel-mikula-sonarsource** [comment]
Hi @jagatgame,

Thank you for reporting this case. We confirm this as a False Positive case.
